pkg/terminal/terminal.go

```diff
@@ -210,22 +211,36 @@ func (t *Term) Run() (int, error) {
 	signal.Notify(ch, syscall.SIGINT)
 	go t.sigintGuard(ch, multiClient)
 
+	fns := trie.New()
+	cmds := trie.New()
+	funcs, _ := t.client.ListFunctions("")
+	for _, fn := range funcs {
+		fns.Add(fn, nil)
+	}
+	for _, cmd := range t.cmds.cmds {
+		for _, alias := range cmd.aliases {
+			cmds.Add(alias, nil)
+		}
+	}
+
 	t.line.SetCompleter(func(line string) (c []string) {
+		prefix := ""
```

Comment:
I think we should look use the code in `Commands.Find` to find out which command it is, like CallWithContext does. And then use the `command` object to decide what to do.

Reply:
I'm not sure what this buys us. `Commands.Find` returns a function object, so I suppose we could type switch on that?

Reply:
I was thinking of changing `Commands.Find` to return the `command` object and then look into that (or make another method that gets called both by `Commands.Find` and here).

Regarding what this buys us is that it will work with user defined aliases instead of a hardcoded list.

Reply:
Ah, I see now. Done.